go/vt/sqlparser/ast_funcs.go

```diff
@@ -480,6 +481,26 @@ func (node *Literal) HexDecode() ([]byte, error) {
 	return hex.DecodeString(node.Val)
 }
 
+// EncodeHexValToMySQLQueryFormat encodes the hexval back into the query format
+// for passing on to MySQL as a bind var
+func (node *Literal) EncodeHexValToMySQLQueryFormat() ([]byte, error) {
```

Comment:
does this really need to be a public method?

Reply:
Good point. And, no, it doesn't. :) Done. Thanks!